src/material/core/focus_indicator/_focus-indicator.scss

```diff
@@ -0,0 +1,54 @@
+@import '../theming/theming';
+
+@mixin mat-focus-indicator() {
+  // Focus indicator base styles.
+  .mat-focus-indicator::before {
+    border-radius: 4px;
+    border-width: 2px;
+    border-style: solid;
+    box-sizing: border-box;
+    position: absolute;
+    pointer-events: none;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+
+  // Focus indicator offsets.
+  .mat-focus-indicator {
+    &.mat-button-base::before,
+    &.mat-card::before,
+    &.mat-calendar-body-cell::before,
+    &.mat-sort-header-button::before {
+      margin: -4px;
+    }
+
+    &.mat-tab-link::before,
+    &.mat-tab-label::before {
+      margin: 4px 0;
+    }
+  }
+
+  // Render pseudo-element on focus.
+  .mat-focus-indicator {
+    // Checkbox, Option, Slide Toggle
+    .cdk-focused.mat-checkbox &::before,
+    .cdk-focused.mat-radio-button &::before,
+    .cdk-focused.mat-slide-toggle &::before,
+
+    // Option
+    &.mat-option.mat-active::before,
```

Comment:
`mat-option` doesn't take focus on keyboard navigation, so can't watch the `:focus` state.

Reply:
I'm not sure if we're consistent about always using the `mat-active` class for active descendant, but if so we could remove the `.mat-option` part of this selector and just have it be a feature if we need to use active descendant for more things in the future